Add Architecture details (#152)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

In order to accommodate as many Transparency Service implementations as possible, this document only specifies the format of Signed Statements (which must be used by all Issuers) and a very thin wrapper format for Receipts, which specifies the Transparency Service identity and the agility parameters for the Signed Inclusion Proofs.
Most of the details of the Receipt's contents are specified in the COSE Signed Merkle Tree Proof document {{-COMETRE}}.

SCITT consists of the following roles: Issuers, Transparency Services, Verifiers, and Auditors.
The process works as follows:

1. Issuers make Statements about Artifacts
1. Issuers sign the Statements, producing Signed Statements
1. Issuers submit the Signed Statements to Transparency Services

> 1. Transparency Services take the submitted Signed Statements and insert them into an Append-only Log, producing Transparent Statements.

Reviewer: ```suggestion
1. Transparency Services evaluate Signed Statements against Registration Policies, counter-signing them if accepted
1. Successfully registered Transparent Statements return Receipts to the issuer
```

> This process can be thought of as Transparency Services counter-signing Issuers' Signed Statements

Reviewer: ```suggestion
```

> 1. Verifiers consume the Transparent Statements, and by verifying them, are convinced of the authenticity and integrity of Statements about an Artifact

Reviewer: ```suggestion
1. Verifiers query a Transparency Service for one or more Transparent Statements associated with a Subject
1. By verifying the Transparent Statement payloads, the verifier may be convinced to the authenticity and integrity of an Artifact
```

Reviewer 2: @SteveLasker Your suggestion at line 387 is not correct. A Verifier need not and should not query the Transparency Service. That is the whole point of the Append only log is that N- Independent Entities (receiving Transparent Statements) need to need to check the log, but just use the Transparent statment possessing the receipt to Verify the presence of the Statement in the log.

Reviewer: @yogeshbdeshpande, I believe we're mixing multiple scenarios here. 
An issuer registers Signed Statements and gets a Receipt. How does a verifier get all the receipts?
A verifier may not have access to the Receipts, and simply wants to get the current status for an artifact. What Statements have been made by Issuers they trust? Have any of the Issuers made amended statements that should be known? By querying the feed, based on the subject, they can see a wholistic picture and decide if the artifact meets their needs.

Reviewer 3: @thgoebel accept or dismiss this suggestion please : )

Reviewer 4: > 1. By verifying the Transparent Statement payloads, the verifier may be convinced to the authenticity and integrity of an Artifact

This means effectively verifying `Statement playload`, which is not a thing. Did you mean `Sign Statement` instead of `Transparent Statement payloads`?

Author: I think @SteveLasker's main concern is having a bullet on how the Verifier obtains the Transparent Statement?
Maybe we can list the different options here. Querying the TS, getting it from the Issuer directly, any others?

Reviewer 4: > I think @SteveLasker's main concern is having a bullet on how the Verifier obtains the Transparent Statement?
Maybe we can list the different options here. Querying the TS, getting it from the Issuer directly, any others?

Okay. But how is that related to the new item?

> 1. By verifying the Transparent Statement payloads, the verifier may be convinced to the authenticity and integrity of an Artifact

Author: This "new" item is just my proposal pulled down and slightly edited ("payload" added, "s/of Statements about an Artifact/of an Artifact"). They are related in that you first need to obtain the Transparent Statement before you can verify it :)

I pushed an update for this.

Reviewer 4: oh! I misread the suggestion scope. Sorry. So my question was targeted at you in the end! I'll chime in again, if this item stalls.

1. Auditors verify the correctness and consistency of the Transparent Statements

> Note that Issuers commit to a Statement by signing it and by submitting it to a Transparency Service.

Reviewer: ```suggestion
Issuers commit to a Statement by signing it and by registering it with a Transparency Service.
```

> They are not necessarily the first ones nor the only ones who publish a Statement.

Reviewer: ```suggestion
They are not necessarily the first ones nor the only ones who register Signed Statements on one or more Transparency Services.
```

Author: Here I intentionally said "Statement" rather than "Signed Statement". If Alice and Bob sign the same statement, the Signed Statements will differ. But what I care about is the Statement that Alice and Bob agree on.

I rephrased this to clarify it.

Reviewer: Thanks, @thgoebel. The nit I was looking to change was "publish" vs. "register". The process by which Signed Statements get added to the Append-only Log is through a registration policy. For consistency, we use "register" as opposed to publish, post, submit, etc. 

A verifier doesn't necessarily need to see agreement between various Issuers. A verifier can choose to trust, and filter in/out specific Issuers. An issuer may trust Alice and Bob, but statements made by Chuck are ignored. Whether they're malicious, or not relevant to the verifiers scenarios aren't overly important as there are scenarios where different issuers can make valid, yet conflicting statements based on a particular scenario or environment.

Reviewer 4: this seems resloved. @SteveLasker could you please resolve, if it is?

> An entity can take on multiple roles.

Reviewer: For the various pieces of feedback above, you may want to split out multiple roles from above. 
Note: we are having a conversation about terminology associated with RATS and other IETF Specs. See #120 and the latest [IETF SCITT Meeting notes](https://datatracker.ietf.org/wg/scitt/meetings/)

Author: Since that discussion is ongoing, I would leave it as-is for now. And update it later in a separate PR :)

> For example, an entity can first act as a Verifier, and later act as an Issuer by making a Statement about the verification outcome.

Author: I added a better example for the "Verifier + Issuer" case, and added your "Verifier + Auditor" case as well.

Issuers MAY use different signing keys (identified by `kid` in the resolved key manifest) for different Artifacts, or sign all Signed Statements under the same key.

### Support for Multiple Statements

An Issuer can make multiple Statements about the same Artifact. For example, an Issuer can make amended Statements about the same Artifact as their view changes over time.

> Multiple Issuers can make different, even conflicting Statements, about the same Artifact. Verifiers can choose which Issuers they trust.

Reviewer: ```suggestion
Multiple Issuers can make different, even conflicting Statements, about the same Artifact.
Verifiers can choose which Issuers they trust.
```

Author: Moved to https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/pull/153.

Still, I prefer not to have a newline here. To me, the sentence "Verifiers can choose which Issuers they trust" is closely linked to the previous "conflicting Statements". When on its own paragraph, this sentence seems to be coming out of the blue, with little context.

Reviewer: Note, the newline is PR feedback friendly, and markdown will put this on the same line.

Author: Ah oops, I thought there was a blank line inbetween. Fixed, added that newline.
